src/signTx.c

```diff
@@ -765,12 +799,19 @@ static void signTx_handleCertificate_ui_runStep()
 
 		case CERTIFICATE_TYPE_STAKE_DELEGATION:
 			snprintf(title, SIZEOF(title), "Delegate stake to pool");
-			encode_hex(
-			        ctx->stageData.certificate.poolKeyHash, SIZEOF(ctx->stageData.certificate.poolKeyHash),
-			        details, SIZEOF(details)
-			);
+			size_t length = encode_hex(
+			                        ctx->stageData.certificate.poolKeyHash, SIZEOF(ctx->stageData.certificate.poolKeyHash),
+			                        details, SIZEOF(details)
+			                );
+			ASSERT(length == strlen(details));
+			ASSERT(length == 2 * SIZEOF(ctx->stageData.certificate.poolKeyHash));
 			break;
 
+		case CERTIFICATE_TYPE_STAKE_POOL_REGISTRATION:
+			snprintf(title, SIZEOF(title), "Register");
+			snprintf(details, SIZEOF(details), "stake pool");
+			UI_STEP_JUMP(HANDLE_CERTIFICATE_STEP_RESPOND);
```

Comment:
does this really work this way? I don't remember how exactly was the macro supposed to be used and I don't like we are formatting something without uiDisplay ...